python/python/benchmarks/test_take.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+import lance
+import numpy as np
+import pyarrow as pa
+import pytest
+from lance import LanceDataset
+
+BATCH_SIZE = 1024
+
+
+def create_dataset(
+    path: str,
+    data_storage_version,
+    num_batches: int,
+    file_size: int,
+    batch_size: int = BATCH_SIZE,
+) -> LanceDataset:
+    schema = pa.schema(
+        [
+            pa.field("i", pa.int32(), nullable=False),
+            pa.field("f", pa.float32(), nullable=False),
+            pa.field("s", pa.binary(), nullable=False),
+            pa.field("fsl", pa.list_(pa.float32(), 2), nullable=False),
+            pa.field("blob", pa.binary(), nullable=False),
+        ]
+    )
+
+    batches = []
+    for i in range(num_batches):
+        start = i * batch_size
+        stop = (i + 1) * batch_size
+        arr_i = pa.array(np.arange(start, stop, dtype=np.int32))
+        arr_f = pa.array(np.arange(start, stop, dtype=np.float32))
+        arr_s = pa.array([f"blob-{x}".encode() for x in range(start, stop)])
+        fsl_values = np.array(
+            [
+                float(batch_size + (x - batch_size) // 2)
+                for x in range(start, stop + batch_size)
+            ],
+            dtype=np.float32,
+        )
+        arr_fsl = pa.FixedSizeListArray.from_arrays(pa.array(fsl_values), 2)
+        arr_blob = pa.array([f"blob-{x}".encode() for x in range(start, stop)])
+
+        batch = pa.RecordBatch.from_arrays(
+            [arr_i, arr_f, arr_s, arr_fsl, arr_blob], schema=schema
+        )
+        batches.append(batch)
+
+    table = pa.Table.from_batches(batches, schema=schema)
+    return lance.write_dataset(
+        table,
+        path,
+        max_rows_per_file=file_size,
+        max_rows_per_group=batch_size,
+        data_storage_version=data_storage_version,
+    )
+
+
+def gen_ranges(total_rows, num_rows):
+    return np.random.choice(total_rows, num_rows, replace=False)
+
+
+@pytest.mark.benchmark()
+@pytest.mark.parametrize("file_size", (1024 * 1024, 1024))
+@pytest.mark.parametrize("lance_format_version", [("2.0", "V2_0"), ("2.1", "V2_1")])
+@pytest.mark.parametrize("num_rows", [1, 10, 100, 1000])
+@pytest.mark.parametrize("batch_size", [512, 1024, 2048])
+@pytest.mark.parametrize("scheme", ["memory", "file"])
```

Comment:
I think you need a prefix, rather than a scheme here, because if we want to test object store it will need to be a storage prefix that includes the bucket name.

Reply:
I tend to place the tests that integrate with the object store into a single test file. Some considered reason:

* Access object store needs to specify credentials, which is hard to parametrize.
* memory and file schemes are out of the box to run the bench, no more information or third-party services. I tend to think the object store belongs to the integration test.

WDYT?

Reply:
that sounds reasonable. I also like the approach of using environment variable so user can set it to run against cloud if necessary and don't need to re-write the function again. I remember we proposed that in the rust benchmark, but probably did not fully merge or finish it. What do you think about doing that approach?